Tasks/Common/azure-arm-rest/azure-arm-resource.ts

```diff
@@ -24,6 +26,45 @@ export class ResourceManagementClient extends azureServiceClient.ServiceClient {
     }
 }
 
+export class Resources {
+    private _client: ServiceClient;
+    
+    constructor(endpoint: AzureEndpoint) {
+        var credentials = new msRestAzure.ApplicationTokenCredentials(endpoint.servicePrincipalClientID, endpoint.tenantID, endpoint.servicePrincipalKey, 
+            endpoint.url, endpoint.environmentAuthorityUrl, endpoint.activeDirectoryResourceID, endpoint.environment.toLowerCase() == 'azurestack');
+        this._client = new ServiceClient(credentials, endpoint.subscriptionID, 30);
+    }
+
+    public async getResources(resourceType: string, resourceName: string) {
+        var httpRequest = new webClient.WebRequest();
+        httpRequest.method = 'GET';
+
+        httpRequest.uri = this._client.getRequestUri('//subscriptions/{subscriptionId}/resources', {},
+        [`$filter=resourceType EQ \'${resourceType}\' AND name EQ \'${resourceName}\'`], '2016-07-01');
+
+        var result = [];
+        try {
+            var response = await this._client.beginRequest(httpRequest);
+            if(response.statusCode != 200) {
+                throw ToError(response);
+            }
+
+            result = result.concat(response.body.value);
+            if(response.body.nextLink) {
+                var nextResult = await this._client.accumulateResultFromPagedResult(response.body.nextLink);
```

Comment:
the second set of results can also have nextLink, right ?

Reply:
accumulateResultFromPagedResult has a loop that queries for the result for next result API and so on.